src/CoreMidi/MidiThruConnectionParams.cs

```diff
@@ -319,32 +321,32 @@ internal void ReadStruct (NSData data)
 
 		internal NSData WriteStruct ()
 		{
-			if (Sources?.Length > 0) {
+			if (Sources?.Length > 0 && connectionParams.Sources is not null) {
 				connectionParams.NumSources = (uint)Sources.Length;
 				for (int i = 0; i < Sources.Length; i++)
-					connectionParams.Sources[i] = Sources[i];
+					connectionParams.Sources [i] = Sources [i];
 			}
```

Comment:
I could possibly see this being a change in behavior since now it's possible that the `connectionParams.NumSources` could not be set if the `connectionParams.Sources is null`. I don't think that we want it to be set in that scenario anyways so either this is negligible, I can add a conditional around the for loop (but it looks messy), or I can just add an `!` saying the `connectionParams.Sources` is not null.

Same thing with the `connectionParams.NumDestinations` a few lines below.